### PR TITLE
GSoC 2019: Add Examples to UnitTest Documentation

### PR DESCRIPTION
**Pull request by krishnadey30**
This PR adds some real examples to Unit Test Documentation.
- [x] asserts
- [x] numLocales
- [x] dependency

Documentation can be found [here](https://krishnadey30.github.io/UnitTest-Docs/).

### CODE REVIEW

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 1]
```diff
@@ -25,13 +25,53 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
    use UnitTest;
+    
+   // calculates factorial
+   proc factorial(x: int): int {
+     return if x == 0 then 1 else x * factorial(x-1); 
+   }
 
    proc test1(test: Test) throws {
-     test.assertTrue(True);
+     test.skipIf(factorial(0) != 1,"Base condtion is wrong in factorial");
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Could we add another test to this example that demonstrates the unconditional `skip` as well?

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 2]
```diff
@@ -25,13 +25,53 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
 
+Another Example
+
+.. code-block:: chapel
+
+   use UnitTest;
+    
+   proc celius2fahrenheit(x) {
+     return (x * 9/5)+32;
+     // we should be returning "(x: real * 9/5)+32"
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Stylistically, I think these comments should come before the line they are commenting.

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 3]
```diff
@@ -25,13 +25,53 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
+   }
+
+   proc test_temperature(test: Test) throws {
+     test.assertFalse(celius2fahrenheit(37) == 98);
+     // we were expecting 98.6 but since we missed typecasting
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Stylistically, I think these comments should come before the line they are commenting.

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 4]
```diff
@@ -40,34 +80,66 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
 
 .. code-block:: chapel
 
-   proc test5(test: Test) throws {
-     test.dependsOn(test3);
+   use UnitTest;
+    
+   var factArray: [1..0] int;
+
+   // calculates factorial
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Stylistically, this should be formatted as a doc-string:

```
/* calculates factorial */

Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: I have made the changes.

File: test/library/packages/UnitTest/UnitTest.chpl (hunk 3 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Updated

File: test/library/packages/UnitTest/UnitTest.chpl (hunk 2 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Updated

File: test/library/packages/UnitTest/UnitTest.chpl (hunk 1 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Added a new example

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 5]
```diff
@@ -22,52 +28,187 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
 
+Examples
+=========
+
+Basic Usage
+-------------
+
+Here is a minimal example demonstrating how to use the UnitTest module:
+
+<celsius2fahrenheit example>
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Sorry, I didn't mean the literal `<example>` text, but rather that as a placeholder for where you should put the example (which is below).

I'd just remove these lines.

Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Sorry, I didn't mean the literal `<example>` text, but rather that as a placeholder for where you should put the example (which is below).

I'd just remove these lines.

Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Updated.

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 6]
```diff
@@ -22,52 +28,187 @@ Class. We use :proc:`~UnitTest.runTest` to pass the tests.
...
+  FAILED failures = 1 
+
+
+Skipping Tests
+---------------
+
+You can skip tests unconditionally with :proc:`~Test.skip` and 
+conditionally with :proc:`~Test.skipIf`:
+
+<factorial example with skips>
```
[the remark this answers is not in the document]
Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: Hey @ben-albrecht I have updated it.

File: test/library/packages/UnitTest/UnitTest.chpl [hunk 7]
```diff
@@ -9,9 +9,15 @@ Any function of the form
...
-*Using Asserts*
+  Tests must be launched with a test launcher. Currently, this is a standalone application 
+  built separately. However, this functionality will eventually be integrated into mason.
+
+Assert Functions
+================
+
+Here are the assert functions available in the UnitTest module:
 
 :proc:`~Test.skip`
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/UnitTest.chpl]: Can we put these in a list? e.g.

```
- :proc:`~Test.skip`
- ....
```

Comment (krishnadey30) [reply test/library/packages/UnitTest/UnitTest.chpl]: I was also thinking of doing that. Thanks

**Pull request merged**